### PR TITLE
[wxwidgets] Add feature wxdebug-level

### LINKED ISSUE #25437: [wxWidgets] wxDEBUG_LEVEL should not be set to 0
**Describe the bug**
After https://github.com/microsoft/vcpkg/commit/6e16931acdf85ab5476973ba00d8eda733f42858 , wxWidgets is broken for dependent projects in Release mode, unless they also add the extra wxDEBUG_LEVEL=0 definition in building.

According to https://docs.wxwidgets.org/3.1.6/group__group__funcmacro__debug.html , the purpose of wxDEBUG_LEVEL is to remove debugging symbols completely, which almost no one does, including wxWidgets' official DLL release and its example project. That's why that commit also needs to [set wxDEBUG_LEVEL manually for examples](https://github.com/microsoft/vcpkg/commit/6e16931acdf85ab5476973ba00d8eda733f42858#diff-c2ba89066671155626d46c4112a763ba06d8ec5abbc7cab8847a2b6b8c2f421fR7).
The proper way to make a release build is to set the `NDEBUG` macro, which deactivates those debugging function, but still keep those symbols and will not cause a linker error. The commit message of 6e16931acdf85ab5476973ba00d8eda733f42858 is misleading, thost debug asserts are NOT active in Release builds. Debugging symbols are there, but they are doing nothing.
This is againt guideline ["Enable existing users of the library to switch to vcpkg"](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#enable-existing-users-of-the-library-to-switch-to-vcpkg), as it sets non-default build option for the library, and requires dependent code also set that option.

**Environment**
- OS: Windows 10.0.19044.1766
- Compiler: Visual Studio 2022 v17.2.5

**To Reproduce**
Run the Github Action of [wangqr/Aegisub](https://github.com/wangqr/Aegisub/) with vcpkg after 6e16931acdf85ab5476973ba00d8eda733f42858

**Expected behavior**
Code build without error

**Failure logs**
[gist](https://gist.github.com/wangqr/0be83aeafb831caa6a68707d334e2587)

### PR DESCRIPTION
**Pull request by Adela0814**
Fix https://github.com/microsoft/vcpkg/issues/25437

Changed the option to port feature.

### CODE REVIEW

File: ports/wxwidgets/portfile.cmake [hunk 1]
```diff
@@ -36,6 +36,12 @@ vcpkg_check_features(
         sound   wxUSE_SOUND
 )
 
+if("wxdebug-level" IN_LIST FEATURES)
+    list(APPEND FEATURE_OPTIONS -DwxBUILD_DEBUG_LEVEL=0)
+else()
+    list(APPEND FEATURE_OPTIONS -DwxBUILD_DEBUG_LEVEL=1)
+endif()
```
Comment (dg0yt) [inline ports/wxwidgets/portfile.cmake:43]: The name `wxdebug-level` doesn't sufficiently indicate that it sets the value to zero if used and to one if not used.
`wxbuild-debug-level-0` would me more expressive.

File: ports/wxwidgets/vcpkg.json [hunk 2]
```diff
@@ -48,6 +48,9 @@
           "platform": "!windows & !osx"
         }
       ]
+    },
+    "wxdebug-level": {
+      "description": "Enable assert checks"
```
Comment (dg0yt) [inline ports/wxwidgets/vcpkg.json:53]: The description is the contrary of what is in the portfile: The presence of the macro disables all remaining debugging hooks.

**Review by dg0yt — CHANGES_REQUESTED** [review]
IMO the current state of this PR is invalid: 
The activation of feature `wxdebug-level` removes API. This is not allowed in vcpkg: Downstream ports could no longer express their dependency.
If anything, the *default* debug level (1) can be tied to a *default* feature.
And the debug level 0 should be burned into the headers in a way that downstream ports follow the `wxwidgets` ports configuration.

File: ports/wxwidgets/vcpkg.json (hunk 2 above)
Comment (Adela0814) [reply ports/wxwidgets/vcpkg.json]: Thanks for your comments, I have updated the description.

File: ports/wxcharts/portfile.cmake [hunk 3]
```diff
@@ -5,10 +5,16 @@ vcpkg_from_github(
...
+else()
+    list(APPEND FEATURE_OPTIONS -DwxBUILD_DEBUG_LEVEL=1)
+endif()
+
 vcpkg_cmake_configure(
     SOURCE_PATH "${SOURCE_PATH}"
-    OPTIONS_RELEASE
-        -DwxBUILD_DEBUG_LEVEL=0
+    OPTIONS
+        ${FEATURE_OPTIONS}
```
Comment (JackBoosY) [inline ports/wxcharts/portfile.cmake:17]:
```suggestion
    OPTIONS_RELEASE
        -DwxBUILD_DEBUG_LEVEL=0
    OPTIONS_DEBUG
        -DwxBUILD_DEBUG_LEVEL=1
```

**Comment by Adela0814** [comment]
Duplicate of https://github.com/microsoft/vcpkg/pull/25572

**Review by dg0yt — CHANGES_REQUESTED** [review]
It is still the wrong approach. If one downstream port requests `wxwidgets[wxdebug-level-0]`, it will remove behaviours for other downstream ports. 

["Features must be treated as additive functionality."](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#do-not-use-features-to-implement-alternatives) 
Wxwidget's debug support is a set of behaviours which are enabled by the recommended default level 1. 
So the natural representation for level 1 is a default feature in vcpkg.
Cf. ["Default features should enable behaviors, not APIs"](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#default-features-should-enable-behaviors-not-apis)

**Comment by AenBleidd** [comment]
@dg0yt, since this is closed, do you have an idea how to implement this feature in a correct way?

**Comment by dg0yt** [comment]
>  since this is closed, do you have an idea how to implement this feature in a correct way?

I outlined this already twice in this PR: There must be default feature to control (i.e. enable )wxwidgets default debugging support.
In that way, you get default behaviour by default, and can opt out on demand.

**Pull request closed without merging**